dev/Scroller/Scroller.cpp

```diff
@@ -4851,6 +4909,38 @@ void Scroller::UpdateUnzoomedExtentAndViewport(
         UpdateScrollControllerValues(ScrollerDimension::VerticalScroll);
     }
 
+    if (horizontalViewportChanged && m_horizontalSnapPoints && m_horizontalSnapPointsNeedViewportUpdates)
+    {
+        // At least one horizontal scroll snap point is not near-aligned and is thus sensitive to the
+        // viewport width. Regenerate and set up all horizontal scroll snap points.
+        winrt::IObservableVector<winrt::ScrollSnapPointBase> horizontalSnapPoints =
+            m_horizontalSnapPoints.try_as<winrt::IObservableVector<winrt::ScrollSnapPointBase>>();
+
+        bool horizontalSnapPointsNeedViewportUpdates = SnapPointsViewportChangedHelper(
+            horizontalSnapPoints,
+            m_viewportWidth);
+        MUX_ASSERT(horizontalSnapPointsNeedViewportUpdates);
```

Comment:
The assert is here because if we hit this code and run the updates we should have at least on that said it needed updates?

Reply:
Yes, this branch is entered only when m_horizontalSnapPointsNeedViewportUpdates is True, and there is no reason why the snap points' answers should be any different from that old evaluation.